src/go/rpk/pkg/cli/connect/install.go

```diff
@@ -0,0 +1,154 @@
+// the latest plugin. Version string let you select a specific version to
+// download, if "latest" or an empty string is passed, it will download the
+// latest version.
+func installConnect(ctx context.Context, fs afero.Fs, version string) (path string, err error) {
+	// We check this before calling the API, to avoid getting the binary
+	// and later fail due to a user setting.
+	pluginDir, err := plugin.DefaultBinPath()
+	if err != nil {
+		return "", fmt.Errorf("unable to determine plugin default path: %v", err)
+	}
+
+	art, _, err := getConnectArtifact(ctx, version)
+	if err != nil {
+		return "", err
+	}
+
+	return downloadAndInstallConnect(ctx, fs, pluginDir, art.Path, art.Sha256)
+}
+
+func getConnectArtifact(ctx context.Context, version string) (connectArtifact, string, error) {
+	plCl, err := newRepoClient()
+	if err != nil {
+		return connectArtifact{}, "", err
+	}
+	manifest, err := plCl.Manifest(ctx)
+	if err != nil {
+		return connectArtifact{}, "", err
+	}
+	var (
+		art        connectArtifact
+		retVersion string
+	)
+	if version == "latest" || version == "" {
+		art, retVersion, err = manifest.LatestArtifact()
+		if err != nil {
+			return connectArtifact{}, "", err
+		}
+	} else {
+		art, err = manifest.ArtifactVersion(version)
+		if err != nil {
+			return connectArtifact{}, "", err
+		}
+		retVersion = version
+	}
+	return art, retVersion, nil
+}
+
+func downloadAndInstallConnect(ctx context.Context, fs afero.Fs, installPath, downloadURL, expShaPrefix string) (string, error) {
+	bin, err := plugin.Download(ctx, downloadURL, true, expShaPrefix)
+	if err != nil {
+		return "", fmt.Errorf("unable to download Redpanda Connect from %q: %v", downloadURL, err)
+	}
+
+	// Ensure the dir exists, but if we have to create it, we do not want
+	// sudo.
+	if exists, _ := afero.DirExists(fs, installPath); !exists {
+		if rpkos.IsRunningSudo() {
+			return "", fmt.Errorf("detected rpk is running with sudo; please execute this command without sudo to avoid saving the plugin as a root owned binary in %s", installPath)
+		}
+		err = os.MkdirAll(installPath, 0o755)
+		if err != nil {
+			return "", fmt.Errorf("unable to create plugin directory %s: %v", installPath, err)
+		}
+	}
+	zap.L().Sugar().Debugf("writing Redpanda Connect plugin to %v", installPath)
+	path, err := plugin.WriteBinary(fs, "connect", installPath, bin, false, true)
+	if err != nil {
+		return "", fmt.Errorf("unable to write Redpanda Connect plugin: %v", err)
+	}
+	return path, nil
+}
+
+// validateVersion validates that the provided version in the flag is either
+// 'latest' or it's a full semantic version.
+func validateVersion(version string) error {
+	// This simple regexp just matches that a semver is in the string, it may
+	// be prefixed with 'v' and contain anything after. Nothing to capture.
+	if version == "latest" {
+		return nil
+	}
+	vMatch := regexp.MustCompile(`^v?\d{1,2}\.\d{1,2}\.\d{1,2}`).MatchString(version)
```

Comment:
feels odd to use mustcompile not as a global

Reply:
i get why it doesn't make a difference in this context i'm just saying 😅

Reply:
Yeah, I thought about that but this function only gets called once at the start of the execution, most common scenario for install is using the default version `latest`, and if that's the case, we don't even get to the part where we compile the Regexp.

Reply:
Yep exactly, it completely makes sense in context: the compile only gets called the one time regardless. my comment was purely on vibes 😂

Reply:
+1 with either, but if you move this to a global, put it in a sync.Once (so there is no hit on startup -- particularly because rpk is invoked via tab completion)